silk/tools/otns_manager.py

```diff
@@ -0,0 +1,529 @@
+      self.extaddr = extaddr
+      self.send_extaddr_event()
+
+  def create_otns_node(self):
+    """Call gRPC client to create a node on server for itself.
+    """
+    if self.node_on_otns:
+      self.logger.debug("Node {:d} already on OTNS while trying to create")
+      return
+    self.logger.debug(
+        "Adding node {:d} to OTNS at ({:d},{:d})".format(
+            self.node_id, self.vis_x, self.vis_y))
+    self.grpc_client.add_node(self.vis_x, self.vis_y, self.node_id)
+    # Temporary measure to prevent most cases of race conditions,
+    # and will be fixed once OTNS decouples gRPC add node command
+    # with extaddr reporting
+    time.sleep(1 if self.source_addr[0] == self.dest_addr[0] else 2)
+    self.send_extaddr_event()
+    self.node_on_otns = True
+
+  def delete_otns_node(self):
+    """Call gRPC client to remove the node on server for itself.
+    """
+    if not self.node_on_otns:
+      self.logger.debug("Node {:d} not on OTNS while trying to delete")
+      return
+    self.logger.debug(
+        "Deleting node {:d} on OTNS ".format(self.node_id))
+    self.grpc_client.delete_node(self.node_id)
+    self.node_on_otns = False
+
+  def update_vis_position(self):
+    """Call gRPC client to update the node's visualization position.
+    """
+    self.logger.debug(
+        "moving node {:d} to OTNS at ({:d},{:d})".format(
+            self.node_id, self.vis_x, self.vis_y))
+    self.grpc_client.move_node(self.vis_x, self.vis_y, self.node_id)
+
+
+class WpantundOtnsMonitor(signal.Subscriber):
+  """OTNS log monitor for wpantund process logs.
+
+  Attributes:
+    node (OtnsNode): OTNS node instance that handles UDP messaging.
+  """
+
+  node = None
+
+  def process_log_line(self, line: str):
+    """Process a single line of wpantund log.
+
+    Args:
+        line (str): line of log to process.
+    """
+    status_match = re.search(RegexType.STATUS.value, line)
+    if status_match:
+      message = status_match.group(2)
+
+      if self.node is not None:
+        extaddr_match = re.search(RegexType.EXTADDR_STATUS.value, message)
+        if extaddr_match:
+          self.node.update_extaddr(int(extaddr_match.group(1), 16))
+        else:
+          event = Event.status_event(message)
+          self.node.send_event(event.to_bytes())
+
+    get_extaddr_info_match = re.search(RegexType.GET_EXTADDR_RES.value, line)
+    if get_extaddr_info_match:
+      extaddr = get_extaddr_info_match.group(1)
+      if self.node is not None:
+        self.node.update_extaddr(int(extaddr, 16))
+
+  def subscribeHandle(self, sender, **kwargs):
+    """Handle messages from Publisher, a wpantund process.
+
+    Args:
+      sender (singal.Publisher): publisher of signal.
+      **kwargs (str): published signal.
+    """
+    line = kwargs["line"]
+    self.process_log_line(line)
+
+
+class OtnsManager(object):
+  """OTNS communication manager for a test case.
+
+  Attributes:
+    dispatcher_host (str): host address of OTNS dispatcher.
```

Comment:
Here does 'dispatcher_host' mean 'server_host'? If yes, let's keep the name same in this class too for easy understanding. Thanks

Reply:
`Dispatcher` is a concept of `OTNS` implementation. We should not expose it to Silk users.